fpga_interchange/examples/tests.cmake

```diff
@@ -266,8 +275,37 @@ function(add_interchange_test)
         )
 
         add_custom_target(test-${family}-${name}-dcp DEPENDS ${dcp})
-        add_dependencies(all-${family}-tests test-${family}-${name}-dcp)
-        add_dependencies(all-${device}-tests test-${family}-${name}-dcp)
+        set(last_target test-${family}-${name}-dcp)
+    endif()
+
+    add_dependencies(all-${device}-tests ${last_target})
+    add_dependencies(all-${family}-tests ${last_target})
+
+    if(output_fasm)
+        if(NOT DEFINED device_family)
+            message(FATAL_ERROR "If FASM output is enabled, the device family must be provided as well!")
+        endif()
+
+        # Output FASM target
+        set(fasm ${CMAKE_CURRENT_BINARY_DIR}/${name}.fasm)
+        add_custom_command(
+            OUTPUT ${fasm}
+            COMMAND
+                ${PYTHON_EXECUTABLE} -mfpga_interchange.fasm_generator
+                    --schema_dir ${INTERCHANGE_SCHEMA_PATH}
+                    --family ${device_family}
+                    ${device_loc}
+                    ${netlist}
+                    ${phys} > ${fasm}
```

Comment:
doing redirection like this tends to cause problems if the FASM generator hits an error, as an empty file will be created and so `make` won't automatically re-run the fasm generator when running the makefile again.

I think the best option here is to add an output file argument to the FASM generator.

Reply:
Right, I'll make the corresponding changes to the FASM generator and update this once the new tag is there